manifests/conf.pp

```diff
@@ -50,9 +51,14 @@
       purge   => $syncsources_purge,
       recurse => $syncsources_recurse,
     }
-    $syncsourceswithtrailingslash = "${syncsources}/"
-  } else {
-    $syncsourceswithtrailingslash = undef
+  }
+
+  validate_absolute_path($syncsourceswithsinglejson)
```

Comment:
Without the modification below, the current code fails because the default is `undef` and not an absolute path.

```puppet
if $syncsourceswithsinglejson != undef {
  validate_absolute_path($syncsourceswithsinglejson)
}
```